### PR TITLE
Upgrade Sentry to 4.1.0

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
- upgrade Sentry dependency to 4.10
- replace `SentryHandler` with the equivalent from the `sentry-jul` package
- added support for turning log statements into breadcrumbs (attached to thread-local based scope)
- added a `debug` property to simplify debugging Sentry integration.

### CODE REVIEW

File: extensions/logging-sentry/deployment/src/test/resources/application-sentry-logger-custom.properties
Comment (bruno-garcia) [inline extensions/logging-sentry/deployment/src/test/resources/application-sentry-logger-custom.properties:5]: typo?
```suggestion
quarkus.log.sentry.minimum-event-level=WARN
```

**Review by gsmet — DISMISSED** [review]
Thanks! Nice contribution! Just a note that this PR will need squashing.

I'll let @ia3andy review it given he's the maintainer of this extension!

Please dismiss this comment once everything is squashed.

File: extensions/logging-sentry/deployment/src/test/resources/application-sentry-logger-custom.properties
Comment (maciejwalkowiak) [reply extensions/logging-sentry/deployment/src/test/resources/application-sentry-logger-custom.properties]: Thanks!

**Comment by geoand** [comment]
Great stuff!

Let's get a review from @ia3andy who is the author of this extension

File: extensions/logging-sentry/deployment/src/test/java/io/quarkus/logging/sentry/SentryLoggerCustomTest.java
Comment (ia3andy) [inline extensions/logging-sentry/deployment/src/test/java/io/quarkus/logging/sentry/SentryLoggerCustomTest.java:27]: Ah cool, we can access the `SentryOptions`!

File: extensions/logging-sentry/runtime/src/main/java/io/quarkus/logging/sentry/SentryHandlerValueFactory.java
Comment (ia3andy) [inline extensions/logging-sentry/runtime/src/main/java/io/quarkus/logging/sentry/SentryHandlerValueFactory.java:57]: It seems we are going back and forth here (join then split). I guess, instead we should do something like this (I haven't tested this code):
```java
sentryConfig.inAppPackages
	.filter(p -> p != null)
	.filter(p -> p.size() != 1 || !Objects.equals(p.get(0), "*"))
	.ifPresent(p -> p.forEach(options::addInAppInclude));
```

**Review by ia3andy — CHANGES_REQUESTED** [review]
This is great @maciejwalkowiak, thanks for this PR!

I will give it a try later, until then, I added a minor change request.

Also, could you squash your PR to 1 commit?

**Comment by ia3andy** [comment]
@maciejwalkowiak also, don't forget to run this command on the modules which have code changes: 
```
mvn process-sources
```

It will make sure the code is well formatted.

File: extensions/logging-sentry/deployment/src/test/java/io/quarkus/logging/sentry/SentryLoggerCustomTest.java
Comment (maciejwalkowiak) [reply extensions/logging-sentry/deployment/src/test/java/io/quarkus/logging/sentry/SentryLoggerCustomTest.java:27]: Yes! We have relaxed a bit the API in 4.x

File: extensions/logging-sentry/runtime/src/main/java/io/quarkus/logging/sentry/SentryHandlerValueFactory.java
Comment (gsmet) [reply extensions/logging-sentry/runtime/src/main/java/io/quarkus/logging/sentry/SentryHandlerValueFactory.java]: Hmmm, given it's runtime code, I would even recommend to get rid of the lambdas entirely.

We try to avoid them in runtime code as they consume more memory.

Comment (ia3andy) [reply extensions/logging-sentry/runtime/src/main/java/io/quarkus/logging/sentry/SentryHandlerValueFactory.java]: yeah, that would be even better without lambda (that's on me as I added them in the first place :) ), @maciejwalkowiak you may also replace the others `ifPresent` under that one.

Don't forget to squash to one commit..

**Comment by maciejwalkowiak** [comment]
@ia3andy thanks for the review. Files are formatted properly now, lambda removed, commits squashed.

**Comment by ia3andy** [comment]
Thanks @maciejwalkowiak I will try it tomorrow :)

**Review by ia3andy — APPROVED** [review]
Great stuff :)

And congratulation on your first Quarkus contribution!!

**Comment by maciejwalkowiak** [comment]
Thanks @ia3andy! There is a potential to improve Quarkus + Sentry integration further, adding a filter that attaches HTTP request data to all Sentry events, we have also added distributed tracing in 4.0. So lets see what future brings 🙂

**Comment by ia3andy** [comment]
@gsmet any idea why the CI gets cancelled?

**Comment by geoand** [comment]
@ia3andy see Guillaume's email. Please cancel CI again, we need the bandwidth for today's release

**Comment by ia3andy** [comment]
> @ia3andy see Guillaume's email on the dev list. Please cancel CI again, we need the bandwidth for today's release. Once that is out, we can restart CI

Ok I missed that mail :)

**Comment by gsmet** [comment]
I restarted the job.

**Comment by ia3andy** [comment]
@gsmet @geoand the `2021-02-09T08:34:06.5131978Z [INFO] Quarkus - RESTEasy Reactive - Deployment ........... FAILURE [03:28 min]` has also been failing on my PR on JDK 11 Windows.

**Comment by geoand** [comment]
I've seen that but can never reproduce it. I'll disable the test

**Comment by gsmet** [comment]
Merged, congrats @maciejwalkowiak for your first Quarkus PR!

**Pull request merged**
